### PR TITLE
Library-166: added white list for jobs in method cron

### PR DESCRIPTION
**Pull request by Menes1337**

### CODE REVIEW

File: src/apis.php [hunk 1]
```diff
@@ -418,6 +409,32 @@ protected function cron()
...
     /**
+     * @param array $cronJobs
+     *
+     * @return bool
+     */
+    protected function containsOnlyAllowedCronJobs(array $cronJobs)
+    {
+        $validCronJobs = true;
+
+        if (empty($cronJobs)) {
```
Comment (Skistims) [inline src/apis.php]: this is unnecessary, as the following logic will lead to the same return value

File: src/apis.php [hunk 2]
```diff
@@ -418,6 +409,32 @@ protected function cron()
...
+            return $validCronJobs;
+        }
+
+        foreach ($cronJobs as $cronJob) {
+            if (
+                empty($cronJob['job_name'])
```
Comment (Skistims) [inline src/apis.php]: why is there an empty check?
The in_array check is sufficient. Is there a case I don't come up with?
I also think you should use the strict mode of in_array

Comment (Menes1337) [reply src/apis.php]: please check src/apis.php line 369 - it seems it happened in the past. Maybe we can restructure and throw the exception in the newly created method.
You are right - will add strict mode for in_array.

File: src/apis.php (hunk 1 above)
Comment (Menes1337) [reply src/apis.php]: you are right, fixed

**Review by Skistims — APPROVED** [review]

**Review by Rekhyt — APPROVED** [review]

**Review by Skistims — APPROVED** [review]

**Pull request merged**